src/Cake.Hosts/CakeHostsAliases.cs

```diff
@@ -37,6 +39,33 @@ public static bool HostsRecordExists(this ICakeContext context, String ipAddress
             return result;
         }
 
+        /// <summary>
+        /// Lista all records in hosts file
+        /// </summary>
+        /// <param name="context">The Cake context.</param>
+        /// <returns>If contains records, return all records</returns>
+        [CakeMethodAlias]
+        public static IEnumerable<HostsFile> GetAllRecords(this ICakeContext context)
+        {
+            var cakeHosts = GetCakeHosts(context);
+            var result = cakeHosts.GetAllRecords();
+
+            if (result.Any())
```

Comment:
I'd rather move all the logging into internal `CakeHosts` class that contains all the implementation to preserve the Cake convention that `*Alias` class is only a thin wrapper around the `*Impl` class.